server/Controllers/GameControllers/snake.js

```diff
@@ -1,12 +1,117 @@
+const { SNAKE } = require('./gameConstants');
+
 class Snake {
-  constructor(playerId, initialPositionX, initialPositionY,
-    initialDirection, snakeColor, initialSnakeLength) {
+  /**
+   * @param  {string} playerId
+   * @param  {Object} initialProperties initial values for snake body and motion.
+   * @param  {Function} eatFood Function to check that snake's next movement will be eating food.
+   * @param  {Array} rivalBody Rival snake's body coordinates
+   * @param  {number} intervalNumber
+   */
+  constructor(playerId, initialProperties, eatFood, rivalBody, intervalNumber) {
     this.ownerId = playerId;
-    this.head = { x: initialPositionX, y: initialPositionY };
-    this.direction = initialDirection;
-    this.color = snakeColor;
-    this.bodyLength = initialSnakeLength;
+    this.head = { x: initialProperties.position.x, y: initialProperties.position.y };
+    this.direction = initialProperties.direction;
+    this.velocity = initialProperties.velocity;
+    this.color = initialProperties.color;
+    this.length = initialProperties.length;
+    this.eatFood = eatFood;
+    this.rivalBody = rivalBody;
+    this.interval = intervalNumber; // interval variable, if snake collides we will clear this interval
+    this.life = SNAKE.LIFE;
+    this.lost = false;
+    this.xAxisVelocity = 0; // the x axis movement of snake in each interval
+    this.yAxisVelocity = 0; // the y axis movement of snake in each interval
     this.bodyCoordinates = [];
+    this.createInitialSnakeBody(); // populate initial body of snake according to given length.
+  }
+
+  /**
+   * This function generates snake body coordinates from the given starting head coordinates
+   */
+  createInitialSnakeBody() {
+    switch (this.direction) {
+      case 'down': {
+        this.yAxisVelocity = this.velocity;
+        break;
+      }
+      case 'up': {
+        this.yAxisVelocity = -this.veloctiy;
+        break;
+      }
+      case 'left': {
+        this.xAxisVelocity = -this.velocity;
+        break;
+      }
+      case 'right': {
+        this.xAxisVelocity = this.velocity;
+        break;
+      }
+      default: {
+        break;
+      }
+    }
+    // populating the snake body coordinates of snake
+    for (let x = 0; x < this.length; x += 1) {
+      this.moveSnakeOneStep();
+    }
+  }
+
+  /**
+   * Moves the snakes one steps according to relative change on x axis or y axis.
+   */
+  moveSnakeOneStep() {
```

Comment:
Create a separate function for every if condition to have more readability because right now it's a giant function difficult to understand.

Reply:
done.